src/ui/windows/TogglDesktop/TogglDesktop/ui/ViewModels/TimelineViewModel.cs

```diff
@@ -374,20 +377,43 @@ public class TimeEntryBlock : ReactiveObject
         [Reactive]
         public bool IsEditViewOpened { get; set; }
 
-        public TimeEntryBlock(string timeEntryId)
+        public bool IsResizable { get; set; }
+
+        private readonly double _hourHeight;
+
+        public TimeEntryBlock(string timeEntryId, int hourHeight)
         {
+            _hourHeight = hourHeight;
             TimeEntryId = timeEntryId;
-            CreateTimeEntryFromBlock = ReactiveCommand.Create(() => AddNewTimeEntry());
             OpenEditView = ReactiveCommand.Create(() => Toggl.Edit(TimeEntryId, false, Toggl.Description));
             CreateTimeEntryFromBlock = ReactiveCommand.Create(AddNewTimeEntry);
         }
 
-        public TimeEntryBlock() : this(null) { }
+        public TimeEntryBlock(int hourHeight) : this(null, hourHeight) { }
 
         private void AddNewTimeEntry()
         {
             TimeEntryId = Toggl.CreateEmptyTimeEntry(Started, Ended);
             OpenEditView.Execute().Subscribe();
         }
+
+        private long ConvertOffsetToTime(double height, DateTime date)
+        {
+            var hours = 1.0 * height / _hourHeight;
+            var dateTime = date.AddHours(hours);
+            return Toggl.UnixFromDateTime(dateTime);
+        }
+
+        public void ChangeStartTime()
+        {
+            Toggl.SetTimeEntryStartTimeStamp(TimeEntryId,
+                ConvertOffsetToTime(VerticalOffset, Toggl.DateTimeFromUnix(Started).Date));

```

Comment:
Have you considered having timestamps / timespans / datetime on the viewmodel instead of px-based properties?

Reply:
Yes, and actually there are both start/end timestamp and verticalOffset/height in the view model. At first I thought it would be nice to avoid converters and work only with px-based properties, but it ended up that time-based properties became necessary at some point. I can try to refactor this and place all the px-based logic in the converters.